pkg/ingester/ingester.go

```diff
@@ -304,6 +315,21 @@ func (i *Ingester) Push(ctx old_ctx.Context, req *client.WriteRequest) (*client.
 	var lastPartialErr error
 	samples := client.FromWriteRequest(req)
 
+	if i.state == ring.JOINING {
+		userID, err := user.ExtractOrgID(ctx)
+		if err != nil {
+			// TODO not sure what to do here
+			return &client.WriteResponse{}, nil
```

Comment:
It shouldn't happen (request will be rejected without a userid), but we should at least `return nil, err` if it does.